### PR TITLE
deploy-to-itunes-connect-with-pilot 1.0.0

### PR DESCRIPTION
**Pull request by e-vision**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [X] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [X] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [X] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [X] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

### CODE REVIEW

**Comment by e-vision** [comment]
This step has the same functionality as the itunesconnect-deliver step but adds the ability to distribute the app to external testers via Fastlane pilot.

**Comment by viktorbenei** [comment]
Sounds awesome, thanks for the PR @e-vision !

@godrei @kdobmayer @trapacska please check!

File: steps/deploy-to-itunes-connect-with-pilot/1.0.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+- opts:
+    description: Path of the PKG file, to be deployed to iTunes Connect
+    summary: ""
+    title: PKG path
+  pkg_path: $BITRISE_PKG_PATH
+- opts:
+    description: 'Action to execute: build or distribute'
+    summary: ""
+    title: Pilot action
+  pilot_action: distribute
```
Comment (godrei) [inline steps/deploy-to-itunes-connect-with-pilot/1.0.0/step.yml]: I think this step is only supposed to distribute an app (based on its title).
Also based on `fastlane pilot -h` it seems there is no `build` command.
I would recommend to remove this input and hardcode the `distribute` action, or am i missing something?

**Comment by godrei** [comment]
Hi @e-vision ,
thanks for the PR, please have a look on comment.

Do you know any reason to keep both `deploy-to-itunes-connect-with-deliver` and `deploy-to-itunes-connect-with-pilot`?

**Comment by e-vision** [comment]
@godrei
I think there is a pilot *upload* and a pilot *distribute* command
possible. In the first case you need to specify an ipa or pkg. So the text
in description is wrong and should read ‘upload' or ‘distribute’ (in stead
of ‘build' and 'distribute'. We use both versions in our flow. One for
internal and the other for external testing.

BTW the commands are not properly documented on the Fastlane website.

@Krisztián
I think we could still use deploy-to-itunes-connect-with-deliver for final
submission for review for store submission.


Regards,

Kees Schouten


Aan Zee Service Design
werk: +31 71 364 11 11
mobiel: +31 6 248 309 54
kees@aanzee.nl - http://www.aanzee.nl

http://www.twitter.com/AanZeeNL


On 22 February 2018 at 11:16:07, Krisztián Gödrei (notifications@github.com)
wrote:

*@godrei* commented on this pull request.
------------------------------

In steps/deploy-to-itunes-connect-with-pilot/1.0.0/step.yml
<https://github.com/bitrise-io/bitrise-steplib/pull/1392#discussion_r169909842>
:

> +inputs:
+- ipa_path: $BITRISE_IPA_PATH
+  opts:
+    description: Path of the IPA file, to be deployed to iTunes Connect
+    summary: ""
+    title: IPA path
+- opts:
+    description: Path of the PKG file, to be deployed to iTunes Connect
+    summary: ""
+    title: PKG path
+  pkg_path: $BITRISE_PKG_PATH
+- opts:
+    description: 'Action to execute: build or distribute'
+    summary: ""
+    title: Pilot action
+  pilot_action: distribute

I think this step is only supposed to distribute an app (based on its
title).
Also based on fastlane pilot -h it seems there is no build command.
I would recommend to remove this input and hardcode the distribute action,
or am i missing something?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/bitrise-io/bitrise-steplib/pull/1392#pullrequestreview-98510140>,
or mute the thread

**Comment by godrei** [comment]
Hi @e-vision ,

sorry for my late response!

1, I have looked into it and it seems:

`pilot` is supposed to manage to upload an ipa file to  iTunes Connect / Test Flight, to do so an ipa:

- needs to be uploaded to iTunes Connect / App Store
- needs to be submitted in iTunes Connect / App Store

then it can be submitted to iTunes Connect / Test Flight

in contrast `deliver` is supposed to upload / submit / release your ipa to  iTunes Connect / App Store, to do so an ipa:

- needs to be uploaded to iTunes Connect / App Store
- needs to be submitted in iTunes Connect / App Store

then it can be released to the iTunes Connect / App Store.

So the tools have overlapping subtasks but they have a different goals.

Can you confirm it?

2, Regarding to the `pilot_action` input: in pilot -v 2.83.0 there is no `build` command at all. 

`fastlane pilot -h` :

```
...
  Commands: (* default)
    add          Adds new external tester(s) to a specific app (if given). This will also add an existing tester to an app.
    builds       Lists all builds for given application
    distribute   Distribute a previously uploaded binary to Apple TestFlight
    export       Exports all external testers to a CSV file
    find         Find tester(s) (internal or external) by their email address
    help       * Display global or [command] help documentation
    import       Create external testers from a CSV file
    list         Lists all registered testers, both internal and external
    remove       Remove external tester(s) by their email address
    upload       Uploads a new binary to Apple TestFlight
...
```

Please remove the input and hardcode the `distribute` command: https://github.com/e-vision/steps-itunesconnect-pilot/blob/master/main.go#L380

**Comment by godrei** [comment]
Closing as we did not receive a reply - @e-vision  please let us know if we can help with anything and we'll reopen this issue!

**Comment by marioot** [comment]
Hi, is this PR still live?
Is there a plan to make pilot available in bitrise?

Thanks

**Comment by godrei** [comment]
Hi, 
this PR is closed as we did not receive an answer.
You can create a feature request for the step: https://discuss.bitrise.io/c/feature-request

**Pull request closed without merging**